c7n/resources/redshift.py

```diff
@@ -799,3 +834,51 @@ def process_resource_set(self, resources, tag_keys):
             arn = self.manager.generate_arn(
                 r['ClusterIdentifier'] + '/' + r['SnapshotIdentifier'])
             client.delete_tags(ResourceName=arn, TagKeys=tag_keys)
+
+
+@RedshiftSnapshot.action_registry.register('revoke-access')
+class RedshiftSnapshotRevokeAccess(BaseAction):
+    """Revokes ability of accounts to restore a snapshot
+
+    :example:
+
+        .. code-block: yaml
+
+            policies:
+              - name: redshift-snapshot-revoke-access
+                resource: redshift-snapshot
+                filters:
+                  - type: cross-account
+                    whitelist:
+                      - 012345678910
+                actions:
+                  - type: revoke-access
+    """
+    permissions = ('redshift:RevokeSnapshotAccess',)
+    schema = type_schema('revoke-access')
+
+    def process_snapshot_set(self, snapshot_set):
+        client = local_session(self.manager.session_factory).client('redshift')
+        for s in snapshot_set:
+            s.setdefault('c7n:CrossAccountViolations', []).extend(
+                self.data.get('accounts', []))
```

Comment:
`accounts` need to be added to schema, else they can never be set. its also quite odd to be modifying the resource here  unconditionally with the policy value when may not even be a  value on the resource, i'd suggest making a local variable.

Reply:
this was... an idea i had to remove specific accounts using a parameter... i opted against including it in the PR but forgot to remove this piece of the code.